internal/k8s/secrets/validation.go

```diff
@@ -28,6 +31,9 @@ const SecretTypeJWK api_v1.SecretType = "nginx.org/jwk"
 // SecretTypeOIDC contains an OIDC client secret for use in oauth flows. #nosec G101
 const SecretTypeOIDC api_v1.SecretType = "nginx.org/oidc"
 
+// SecretTypeHtpasswd contains an htpasswd file for use in HTTP Basic authorization.. #nosec G101
+const SecretTypeHtpasswd api_v1.SecretType = "nginx.org/htpasswd"
```

Comment:
Yeah that's the idea. I considered using keys of the secret as usernames and their value as auth string, but this would add processing and the user wouldn't be able to use existing passwd databases with e.g. comments.

Reply:
My thought here is the experience of creating the htpasswd file and how that feels bespoke to a Kubernetes (yaml focused) operator.
I understand the choice though.  The file is both native to NGINX configuration and not unusual in itself in the full context.
Documentation might help me here - walking through defining, generating, and using the htpasswd as part of the configuration.
We have tutorials or examples that could be used for that.

Reply:
There's a gosec linting error, but it's a false positive. We need to skip the error similar to other places. 
```suggestion
const SecretTypeHtpasswd api_v1.SecretType = "nginx.org/htpasswd" // #nosec G101
```